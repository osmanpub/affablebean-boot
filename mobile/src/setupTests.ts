import {configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-enzyme';
import {JSDOM} from 'jsdom';
import 'react-native';

/**
 * Set up DOM in node.js environment for Enzyme to mount to
 */
const jsdom = new JSDOM('<!doctype html><html><body></body></html>');
const {window} = jsdom;

function copyProps(src: any, target: any) {
  Object.defineProperties(target, {
    ...Object.getOwnPropertyDescriptors(src),
    ...Object.getOwnPropertyDescriptors(target),
  });
}
// @ts-ignore
global.window = window;

// @ts-ignore
global.document = window.document;

// @ts-ignore
global.navigator = {
  userAgent: 'node.js',
};

// @ts-ignore
global.requestAnimationFrame = function(callback) {
  return setTimeout(callback, 0);
};

// @ts-ignore
global.cancelAnimationFrame = function(id) {
  clearTimeout(id);
};

copyProps(window, global);

/**
 * Set up Enzyme to mount to DOM, simulate events,
 * and inspect the DOM in tests.
 */
configure({adapter: new Adapter()});

/**
 * Mock components and modules
 */

jest.mock('./net/cart');
jest.mock('./net/category');
jest.mock('./net/categories');
jest.mock('./net/checkout');
