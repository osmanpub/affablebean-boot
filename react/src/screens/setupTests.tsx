import { configureStore } from "@reduxjs/toolkit";
import React from "react";
import ReactDOM from "react-dom";
import ReactTestUtils, { act } from "react-dom/test-utils";
import { Provider } from "react-redux";
import { BrowserRouter as Router, Route } from "react-router-dom";
import reducer from "../redux";
import Cart from "./Cart";
import CategoryProducts from "./CategoryProducts";
import Checkout from "./Checkout";
import Contact from "./Contact";
import Home from "./Home";
import Privacy from "./Privacy";

jest.mock("../net/cart");
jest.mock("../net/category");
jest.mock("../net/categories");
jest.mock("../net/checkout");

export let container: any;

export const setupAll = (callback?: Function) =>
  beforeAll(() => {
    container = document.createElement("div");
    document.body.appendChild(container);

    act(() => {
      const store = configureStore({
        reducer
      });

      const home = (
        <Provider store={store}>
          <Router>
            <Route path="/" exact component={Home} />
            <Route path="/category/:id" component={CategoryProducts} />
            <Route path="/checkout" component={Checkout} />
            <Route path="/contact" component={Contact} />
            <Route path="/privacy" component={Privacy} />
            <Route path="/viewCart/:clear" component={Cart} />
          </Router>
        </Provider>
      );

      ReactDOM.render(home, container);
    });

    callback && callback();
  });

export const changeValue = (widget: any, value: any) => {
  widget.value = value;
  ReactTestUtils.Simulate.change(widget);
};

export const mouseClick = (widget: any) =>
  act(() => {
    widget.dispatchEvent(new MouseEvent("click", { bubbles: true }));
  });

export const afterAll = () => {
  document.body.removeChild(container);
  container = null;
};
