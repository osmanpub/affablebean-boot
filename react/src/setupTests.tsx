import "@testing-library/jest-dom/extend-expect";
import ReactTestUtils, { act } from "react-dom/test-utils";

// Fixes problem - https://stackoverflow.com/questions/48809753/testing-mutationobserver-with-jest
// @ts-ignore
global.MutationObserver = class {
  // @ts-ignore
  constructor(callback) {}
  // @ts-ignore
  disconnect() {}
  // @ts-ignore
  observe(element, initObject) {}
};

export const changeValue = (widget: any, value: any) => {
  widget.value = value;
  ReactTestUtils.Simulate.change(widget);
};

export const mouseClick = (widget: any) =>
  act(() => {
    widget.dispatchEvent(new MouseEvent("click", { bubbles: true }));
  });
