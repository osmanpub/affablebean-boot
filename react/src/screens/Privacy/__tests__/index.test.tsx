import { container, mouseClick, setupAll } from "../../../setupTests";

setupAll(() => {
  // privacy and contact buttons
  const footerButtons = container.querySelectorAll("button.btn.btn-link");
  expect(footerButtons.length).toBe(2);

  // click on privacy
  mouseClick(footerButtons[0]);
});

it("load privacy page", () => {
  const intro = container.querySelector("p");
  expect(intro.textContent).toContain(
    "AffableBean Co knows that you care how information about you is used and shared and we appreciate your trust in us to do that carefully and sensibly. This notice describes the privacy policy of AffableBean Co."
  );
});
