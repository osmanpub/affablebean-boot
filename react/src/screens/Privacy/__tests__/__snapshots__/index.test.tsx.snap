// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Privacy /> renders correctly 1`] = `
<div>
  <div
    className="sc-AxgMl kehYEp"
  >
    <div
      className="sc-AxheI jbKZFY"
    >
      <div
        className="sc-AxiKw byMOZe"
      >
        <div
          className="sc-AxhCb kneLnw headerWidget"
        >
          <a
            className="bubble"
            data-cy="proceed-checkout"
            href="/checkout"
            onClick={[Function]}
          >
            proceed to checkout ➟
          </a>
        </div>
        <div
          className="sc-AxhUy dXnRMF headerWidget"
        >
          <span
            className="horizontalMargin"
            data-cy="cart-hdr-total"
            data-testid="cart-hdr-total"
          >
            <img
              alt="shopping cart icon"
              src="/static/img/cart.gif"
            />
             
          </span>
          <a
            className="bubble"
            data-cy="cart-hdr-view-cart"
            href="/viewCart/false"
            onClick={[Function]}
          >
            view cart
          </a>
        </div>
      </div>
      <div
        className="sc-Axmtr cYyhem"
      >
        <a
          data-cy="home-logo"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="Affable Bean logo"
            src="/static/img/logo.png"
          />
        </a>
      </div>
      <img
        alt="the affable bean"
        id="logoText"
        src="/static/img/logoText.png"
      />
    </div>
  </div>
  <div
    className="singleColumn"
  >
    <p
      data-cy="privacy-intro"
      data-testid="privacy-intro"
    >
      AffableBean Co knows that you care how information about you is used and shared and we appreciate your trust in us to do that carefully and sensibly. This notice describes the privacy policy of AffableBean Co.
       
      <b>
        By visiting AffableBean Co, you are accepting and consenting to the practices described in this Privacy Notice.
      </b>
    </p>
    <br />
    <div
      id="pr2"
    >
      <p>
        <b>
          Controllers of Personal Information
        </b>
      </p>
      <p>
        Any personal information provided to or gathered by AffableBean Co is controlled by AffableBean Global (the data controllers). The Affable Bean Company Limited, Registered office: AffableBean House, Roast Drive, Pothead, POT123, BeanieLand
      </p>
    </div>
    <br />
    <div
      id="pr3"
    >
      <p>
        <b>
          What Personal Information About Customers Does AffableBean Co Gather?
        </b>
      </p>
      <p>
        The information we learn from customers helps us personalise and continually improve your shopping experience at AffableBean Co. We use the information to handle orders, deliver products and services, process payments, communicate with you about orders, products, services and promotional offers, update our records and generally maintain your accounts with us, display content such as wish lists and customer reviews and recommend merchandise and services that might be of interest to you. We also use this information to improve our store and platform, prevent or detect fraud or abuses of our website and enable third parties to carry out technical, logistical or other functions on our behalf.
      </p>
      <p>
        Here are the types of information we gather.
      </p>
      <ul>
        <li>
          <b>
            Information You Give Us:
          </b>
           we receive and store any information you enter on our website or give us in any other way. You can choose not to provide certain information but then you might not be able to take advantage of many of our features. We use the information that you provide for such purposes as responding to your requests, customising future shopping for you, improving our stores, and communicating with you.
        </li>
        <li>
          <b>
            Automatic Information:
          </b>
           we receive and store certain types of information whenever you interact with us. For example, like many websites, we use "cookies" and we obtain certain types of information when your Web browser accesses AffableBean Co or advertisements and other content served by or on behalf of AffableBean Co on other Web sites.
           
        </li>
        <li>
          <b>
            Mobile:
          </b>
           When you download or use apps created by AffableBeanor its subsidiaries and, where applicable, have requested or consented to location services, we may receive information about your location and your mobile device, including a unique identifier for your device. We may use this information to provide you with location-based services, such as search results, and other personalised content. Most mobile devices allow you to turn off location services.
           
        </li>
        <li>
          <b>
            E-mail Communications:
          </b>
           to help us make e-mails more useful and interesting, we often receive a confirmation when you open e-mail from AffableBean Co if your computer supports such capabilities. We also compare our customer list to lists received from other companies in an effort to avoid sending unnecessary messages to our customers. If you do not want to receive e-mail or other mail from us, please adjust your Customer Communication Preferences
        </li>
        <li>
          <b>
            Information from Other Sources:
          </b>
           we might receive information about you from other sources and add it to our account information.
           
        </li>
      </ul>
    </div>
    <br />
    <div
      id="pr4"
    >
      <p>
        <b>
          What About Cookies?
        </b>
      </p>
      <p>
        To enable our systems to recognise your device and to provide features to you, we use cookies.
      </p>
    </div>
    <br />
    <div
      id="pr5"
    >
      <p>
        <b>
          Does AffableBean Co share the Information it receives?
        </b>
      </p>
      <p>
        Information about our customers is an important part of our business and we are not in the business of selling it to others. AffableBean Co shares customer information only as described below and with AffableBean.com, Inc. and the subsidiaries which AffableBean.com, Inc., controls and that are either subject to this Privacy Notice or follow practices at least as protective as those described in this Privacy Notice.
      </p>
      <ul>
        <li>
          <b>
            Affiliated Businesses We Do Not Control:
          </b>
            We work closely with affiliated businesses. In some cases, such as Marketplace sellers, these businesses operate stores at AffableBean Co or sell offerings to you at AffableBean Co. In other cases, we operate stores, provide services or sell product lines jointly with or on behalf of these businesses.
           
        </li>
        <li>
          <b>
            Third Party Service Providers:
          </b>
            We employ other companies and individuals to perform functions on our behalf. Examples include fulfilling orders, delivering packages, sending postal mail and e-mail, removing repetitive information from customer lists, analysing data, providing marketing assistance, providing search results and links (including paid listings and links), processing credit card payments and providing customer service. They have access to personal information needed to perform their functions, but may not use it for other purposes. Further, they must process the personal information in accordance with this Privacy Notice and as permitted by applicable data protection laws.
        </li>
        <li>
          <b>
            Promotional Offers:
          </b>
            Sometimes we send offers to selected groups of AffableBean Co customers on behalf of other businesses. When we do this, we do not give that business your name and address. If you do not want to receive such offers, please adjust your Customer Communication Preferences.
        </li>
        <li>
          <b>
            Business Transfers:
          </b>
            As we continue to develop our business, wemight sell or buy stores, subsidiaries or business units. In such transactions, customer information generally is one of the transferred business assets but remains subject to the promises made in any pre-existing Privacy Notice (unless, of course, the customer consents otherwise). Also, in the unlikely event that AffableBean.com, Inc. or substantially all of its assets are acquired, customer information will of course be one of the transferred assets.
        </li>
        <li>
          <b>
            Protection of AffableBean Co and Others:
          </b>
            We release account and other personal information when we believe release is appropriate to comply with the law; enforce or apply our Conditions of Use and other agreements; or protect the rights, property or safety of AffableBean Co, our users or others. This includes exchanging information with other companies and organisations for fraud protection and credit risk reduction. Obviously, however, this does not include selling, renting, sharing or otherwise disclosing personally identifiable information from customers for commercial purposes in a way that is contrary to the commitments made in this Privacy Notice.
        </li>
        <li>
          <b>
            With your consent:
          </b>
            other than as set out above, you will receive notice when information about you might go to third parties and you will have an opportunity to choose not to share the information.
        </li>
      </ul>
    </div>
    <br />
    <div
      id="pr6"
    >
      <p>
        <b>
          How Secure Is Information About Me?
        </b>
      </p>
      <p>
        We work to protect the security of your information during transmission by using Secure Sockets Layer (SSL) software, which encrypts information you input.We reveal only the last four digits of your credit card numbers when confirming an order. Of course, we transmit the entire credit card number to the appropriate credit card company during order processing. We maintain physical, electronic and procedural safeguards in connection with the collection, storage and disclosure of personally identifiable customer information. Our security procedures mean that we may occasionally request proof of identity before we disclose personal information to you. It is important for you to protect against unauthorised access to your password and to your computer. Be sure to sign off when you finish using a shared computer.
      </p>
    </div>
    <br />
    <div
      id="pr7"
    >
      <p>
        <b>
          What About Third-Party Advertisers and Links to Other Web sites?
        </b>
      </p>
      <p>
        Our site may include third-party advertising and links to other Web sites.
      </p>
    </div>
    <br />
    <div
      id="pr8"
    >
      <p>
        <b>
          What Information Can I Access? 
        </b>
      </p>
      <p>
        AffableBean Co gives you access to a broad range of information about your account and your interactions with AffableBean Co for the limited purpose of viewing and, in certain cases, updating that information.
      </p>
    </div>
    <br />
    <div
      id="pr9"
    >
      <p>
        <b>
          What Choices Do I Have? 
        </b>
      </p>
      <p>
        As discussed above, you can always choose not to provide information, even though it might be needed to make a purchase or to take advantage of such AffableBean Co features as Your Profile, Wish Lists, and Customer Reviews. When you update information, we usually keep a copy of the previous version for our records. If you do not want to receive e-mail or other mail from us, please adjust your Customer Communication Preferences. (If you do not want to receive Conditions of Use and Sale and other legal notices from us, such as this Privacy Notice, those notices will still govern your use of AffableBean Co and orders placed with AffableBean Co, and it is your responsibility to review them for changes.) If you do not want us to use personal information that we gather to allow third parties to personalise advertisements we display to you, please adjust your Advertising Preferences.
      </p>
    </div>
    <br />
    <div
      id="pr10"
    >
      <p>
        <b>
          Conditions of Use, Notices, and Revisions 
        </b>
      </p>
      <p>
        If you have any concern about privacy at AffableBean Co, please e-mail us with a thorough description and we will try to resolve the issue for you. Our business changes constantly and our Privacy Notice and the Conditions of Use and Sale will change also. We may e-mail periodic reminders of our notices and conditions, unless you have instructed us not to, but you should check our website frequently to see recent changes. Unless stated otherwise, our current Privacy Notice applies to all information that we have about you and your account. We stand behind the promises we make, however, and will never materially change our policies and practices to make them less protective of customer information collected in the past without the consent of affected customers.
      </p>
    </div>
  </div>
  <div
    className="sc-AxjAm jZHPdI"
  >
    <br />
    <span
      className="sc-AxirZ bSLrDO"
    >
      <a
        data-cy="privacy"
        data-testid="footer-privacy"
        href="/privacy"
        onClick={[Function]}
      >
        <button
          className="btn btn-link"
        >
          Privacy
        </button>
      </a>
      <a
        data-cy="contact"
        data-testid="footer-contact"
        href="/contact"
        onClick={[Function]}
      >
        <button
          className="btn btn-link"
        >
          Contact
        </button>
      </a>
       © 2020 the affable bean company
    </span>
  </div>
</div>
`;
